app/src/main/java/com/sunasterisk/food_01/data/source/repository/CategoryRepository.kt

```diff
@@ -0,0 +1,25 @@
+package com.sunasterisk.food_01.data.source.repository
+
+import com.sunasterisk.food_01.data.model.Category
+import com.sunasterisk.food_01.data.source.data_source.CategoryDataSource
+import com.sunasterisk.food_01.data.source.local.CategoryLocalDataSource
+import com.sunasterisk.food_01.data.source.remote.CategoryRemoteDataSource
+import com.sunasterisk.food_01.data.source.remote.OnFetchDataJsonListener
+
+class CategoryRepository private  constructor(private val remote: CategoryDataSource.Remote,
+                                              private val local: CategoryDataSource.Local) {
+
+    private object Holder {
+        val INSTANCE = CategoryRepository(
+            remote = CategoryRemoteDataSource.instance,
+            local = CategoryLocalDataSource.instance)
+    }
+
+    fun getCategory(listener: OnFetchDataJsonListener<MutableList<Category>>) {
+        remote.getCategory(listener)
+    }
+
+    companion object {
+        val instance: CategoryRepository by lazy { Holder.INSTANCE }
```

Comment:
Đối với class thiết kế theo singleton mà hàm constructor có tham số truyền vào 
nên viết hàm getInstance có tham số tuyền vào tương ứng, tránh việc khởi tạo đối tượng của lớp khác trong class của mình => khó test
viết theo dạng dưới đây 
```suggestion
    companion object {
        private var instance: UserLocalDataSource? = null
        fun getInstance(dao: Dao) = instance ?: synchronized(this) {
            instance ?: UserLocalDataSource(dao).also { instance = it }
        }
    }
```